# mayhem_program::instructions::pump_interface::buy::handler  (yours 8424 B at .text+0x1ba80; original 13848 B at .text+0x1bae8)
# distance 1993 instructions; evidence: called-by mayhem_program::__private::__global::buy#b519; same-callees 20/26
--- original
+++ yours
@@ -1,260 +1,296 @@
-mov64 r8, r5
 mov64 r7, r4
-stxdw [r10-1256], r3
-stxdw [r10-1248], r2
-mov64 r9, r1
-ldxdw r6, [r9+8]
-ldxdw r2, [r6+8]
-ldxdw r1, [r6+16]
-add64 r1, 8
-add64 r2, 8
-call orig_fn_2dbf8
-jne r0, 2, +0x480
-ldxdw r1, [r6+72]
-ldxdw r2, [r1+96]
-ldxdw r3, [r6+8]
-ldxdw r1, [r6+16]
-stdw [r10-1216], 0
-stdw [r10-1224], 0
-stdw [r10-1232], 0
-stdw [r10-1240], 0
-add64 r1, 8
-add64 r3, 8
-mov64 r4, r10
-add64 r4, -1240
-call orig_fn_2e0a0
-jne r0, 2, +0x480
-ldxdw r1, [r6+72]
-stxdw [r10-1280], r1
-ldxdw r1, [r1+72]
-stxdw [r10-1272], r1
-jeq r1, 0, +0x3478
-stxdw [r10-1320], r9
-stxdw [r10-1304], r8
-stxdw [r10-1288], r6
+mov64 r8, r3
+mov64 r6, r2
+ldxdw r9, [r1+8]
+ldxdw r2, [r9+8]
+ldxb r2, [r2+253]
+jeq r2, 0, +0xc8
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[00000000]
+call mayhem_program::MayhemError::name#2984
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 0
+call <T as alloc::string::SpecToString>::spec_to_string#2731
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 56
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6000
+ja +0x418
+stxdw [r10-2808], r5
+stxdw [r10-2816], r1
+mov64 r1, r10
+add64 r1, -1128
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-1128]
+jne r1, 0, +0x1f28
+stxdw [r10-2800], r6
+ldxdw r1, [r10-1088]
+ldxdw r2, [r9+16]
+ldxdw r3, [r2+88]
+jeq r3, 0, +0x228
+ldxdw r4, [r9+8]
+ldxdw r6, [r4+237]
+jeq r6, 0, +0x228
+mov64 r0, 1
+mov64 r4, 1
+jslt r6, 0, +0x340
+mov64 r4, 0
+mov64 r5, r3
+add64 r5, r6
+jsge r5, r3, +0x358
+xor64 r4, r0
+and64 r4, 1
+jne r4, 0, +0x378
+jsge r1, r5, +0x228
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[0f000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 15
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 60
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6015
+ja +0x418
+stxdw [r2+88], r1
+mov64 r1, r10
+add64 r1, -1128
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-1128]
+jne r1, 0, +0x1f88
+ldxdw r2, [r10-1088]
+ldxdw r1, [r9+16]
+ldxdw r3, [r1+80]
+jsge r2, r3, +0x390
+ldxb r2, [r1+104]
+jeq r2, 1, +0x440
+ldxdw r2, [r9+8]
+ldxdw r2, [r2+144]
+ldxdw r3, [r9+72]
+ldxdw r3, [r3+96]
+jge r3, r2, +0x438
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[030000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 3
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 64
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6003
+ja +0x418
+mov64 r5, r3
+add64 r5, r6
+jslt r5, r3, +0x178
+mov64 r0, 0
+xor64 r4, r0
+and64 r4, 1
+jeq r4, 0, +0x190
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:60:41)
+call core::panicking::panic_const::panic_const_add_overflow#3493
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[04000000]
+call mayhem_program::instructions::pump_interface::sell::SellError::name#46b1
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 4
+call <T as alloc::string::SpecToString>::spec_to_string#d382
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 62
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6004
+mov64 r1, r10
+add64 r1, -1128
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+exit
+stb [r1+104], 1
+ldxdw r1, [r9+72]
+ldxdw r2, [r1+72]
+stxdw [r10-2824], r2
+jeq r2, 0, +0x1368
+stxdw [r10-2840], r9
+stxdw [r10-2848], r1
+ldxdw r1, [r1+80]
+stxdw [r10-2832], r1
 and64 r7, 65535
-ldxdw r1, [r10-1256]
+mov64 r1, r8
 mov64 r2, 0
 mov64 r3, r7
 mov64 r4, 0
 call __multi3
 mov64 r9, r0
 mov64 r6, r1
-ldxdw r1, [r10-1248]
+ldxdw r1, [r10-2800]
 mov64 r2, 0
 mov64 r3, r7
 mov64 r4, 0
 call __multi3
+mov64 r2, r1
+add64 r2, r9
+mov64 r4, 1
+mov64 r3, 1
+jne r6, 0, +0x538
+mov64 r3, 0
+jge r2, r1, +0x540
+or64 r3, r4
+jne r3, 1, +0x558
+ja +0x1368
+jlt r2, r1, +0x520
+mov64 r4, 0
+or64 r3, r4
+jeq r3, 1, +0x1368
+mov64 r1, r0
+mov64 r3, 10000
+mov64 r4, 0
+call __udivti3
 mov64 r7, r0
-mov64 r8, r1
-add64 r8, r9
-mov64 r2, 1
-mov64 r9, 1
-jne r6, 0, +0x1b0
-mov64 r9, 0
-jlt r8, r1, +0x1c0
-mov64 r2, 0
-or64 r9, r2
-ldxdw r1, [r10-1280]
-ldxdw r1, [r1+80]
-stxdw [r10-1312], r1
+mov64 r9, r1
+mov64 r1, 1
+mov64 r6, 1
+ldxdw r2, [r10-2800]
+jge r2, r7, +0x6a8
+mov64 r6, 0
+jlt r8, r9, +0x6b0
+jeq r8, r9, +0x5c8
+mov64 r6, r1
+ldxdw r1, [r10-2832]
 mov64 r2, 0
 lddw r3, 0x38d7ea4c68000
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
 mov64 r1, r0
-ldxdw r3, [r10-1272]
+ldxdw r3, [r10-2824]
 mov64 r4, 0
 call __udivti3
-stxdw [r10-1296], r0
-stxdw [r10-1264], r1
-and64 r9, 1
-jne r9, 0, +0x3478
-mov64 r1, r7
-mov64 r2, r8
-mov64 r3, 10000
-mov64 r4, 0
-call __udivti3
+and64 r6, 1
+jeq r6, 0, +0x1368
+ldxdw r6, [r10-2800]
+mov64 r4, r6
+add64 r4, r7
 mov64 r3, 1
 mov64 r2, 1
-ldxdw r4, [r10-1248]
-jlt r4, r0, +0x2e8
+jlt r4, r6, +0x6c8
 mov64 r2, 0
-ldxdw r8, [r10-1288]
-ldxdw r7, [r10-1256]
-jge r7, r1, +0x300
-mov64 r9, 2
-jeq r7, r1, +0x318
+mov64 r4, r8
+add64 r4, r9
+add64 r4, r2
+jge r4, r8, +0x6e8
+jeq r4, r8, +0x6f8
+mov64 r2, r3
+jne r2, 1, +0x700
+ja +0x1368
+jge r8, r9, +0x5b8
+mov64 r1, 0
+jne r8, r9, +0x5c0
+ja +0x5c8
+mov64 r4, r8
+add64 r4, r9
+add64 r4, r2
+jlt r4, r8, +0x688
+mov64 r3, 0
+jne r4, r8, +0x690
+jeq r2, 1, +0x1368
+mov64 r3, r7
+mov64 r2, 1
+mov64 r4, 1
+jlt r6, r3, +0x728
+mov64 r4, 0
+mov64 r5, r8
+sub64 r5, r9
+sub64 r5, r4
+sub64 r6, r7
+mov64 r4, 1
+jlt r0, r6, +0x7b0
+mov64 r4, 0
+mov64 r6, 1
+jge r1, r5, +0x7c0
+jeq r1, r5, +0x7d0
+mov64 r4, r6
+ldxdw r5, [r10-2800]
+add64 r7, r5
+jge r7, r3, +0x7e8
+and64 r4, 1
+jeq r4, 0, +0x800
+ja +0x858
+mov64 r6, 1
+jlt r1, r5, +0x770
+mov64 r6, 0
+jne r1, r5, +0x778
+ldxdw r5, [r10-2800]
+add64 r7, r5
+jlt r7, r3, +0x798
+mov64 r2, 0
+and64 r4, 1
+jne r4, 0, +0x858
+add64 r9, r8
+add64 r9, r2
+mov64 r3, 1
+mov64 r2, 1
+jle r0, r7, +0x870
+mov64 r2, 0
+jgt r1, r9, +0x878
+jeq r1, r9, +0x888
 mov64 r2, r3
 and64 r2, 1
-jne r2, 0, +0x328
-ja +0x488
-ldxdw r8, [r10-1288]
-ldxdw r7, [r10-1256]
-jlt r7, r1, +0x2b8
+jne r2, 0, +0x898
+mov64 r1, 12
+call mayhem_program::instructions::pump_interface::sell::<impl core::convert::From<mayhem_program::instructions::pump_interface::sell::SellError> for anchor_lang::error::Error>::from#5915
+ja +0x430
+jle r1, r9, +0x838
 mov64 r3, 0
-mov64 r9, 2
-jne r7, r1, +0x2c8
+jne r1, r9, +0x840
 and64 r2, 1
-jeq r2, 0, +0x488
-stxw [r10-24], r9
-mov64 r1, r10
-add64 r1, -304
-mov64 r6, r10
-add64 r6, -24
-mov64 r2, r6
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-56], 0
-stdw [r10-64], 1
-stdw [r10-72], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-144], r1
-mov64 r1, r10
-add64 r1, -72
-stxdw [r10-152], r1
-lddw r1, 0xe0000020
-stxdw [r10-136], r1
-mov64 r2, r10
-add64 r2, -152
+jeq r2, 0, +0x858
+ldxdw r1, [r10-2808]
+and64 r1, 65535
+jgt r1, 10000, +0x20a0
+ldxdw r7, [r10-2840]
+ldxdw r1, [r7+32]
+ldxdw r6, [r1+104]
 mov64 r1, r6
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-jne r0, 0, +0x3488
-ldxdw r1, [r10-56]
-stxdw [r10-1168], r1
-ldxdw r1, [r10-64]
-stxdw [r10-1176], r1
-ldxdw r1, [r10-72]
-stxdw [r10-1184], r1
-ldxdw r1, [r10-304]
-stxdw [r10-1208], r1
-ldxdw r1, [r10-296]
-stxdw [r10-1200], r1
-ldxdw r1, [r10-288]
-stxdw [r10-1192], r1
-or64 r9, 6000
-stxw [r10-1088], r9
-stb [r10-1160], 2
-stdw [r10-1240], 2
-mov64 r1, r10
-add64 r1, -1240
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-exit
-ldxdw r5, [r10-1248]
-mov64 r4, r5
-add64 r4, r0
-mov64 r3, 1
-mov64 r2, 1
-jlt r4, r5, +0x508
 mov64 r2, 0
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jge r4, r7, +0x528
-jeq r4, r7, +0x538
-mov64 r2, r3
-and64 r2, 1
-jeq r2, 0, +0x548
-ja +0x328
-mov64 r4, r7
-add64 r4, r1
-add64 r4, r2
-jlt r4, r7, +0x4e0
-mov64 r3, 0
-jne r4, r7, +0x4e8
-and64 r2, 1
-jne r2, 0, +0x328
-mov64 r3, r0
-mov64 r2, 1
-mov64 r4, 1
-ldxdw r5, [r10-1248]
-jlt r5, r3, +0x578
-mov64 r4, 0
-mov64 r5, r7
-sub64 r5, r1
-sub64 r5, r4
-ldxdw r6, [r10-1248]
-sub64 r6, r0
-mov64 r4, 1
-ldxdw r9, [r10-1296]
-jlt r9, r6, +0x628
-mov64 r4, 0
-mov64 r6, 1
-ldxdw r9, [r10-1264]
-jge r9, r5, +0x640
-ldxdw r9, [r10-1264]
-jeq r9, r5, +0x658
-mov64 r4, r6
-ldxdw r5, [r10-1248]
-add64 r0, r5
-jge r0, r3, +0x670
-mov64 r9, 12
-and64 r4, 1
-jeq r4, 0, +0x690
-ja +0x328
-mov64 r6, 1
-ldxdw r9, [r10-1264]
-jlt r9, r5, +0x5d8
-mov64 r6, 0
-ldxdw r9, [r10-1264]
-jne r9, r5, +0x5e8
-ldxdw r5, [r10-1248]
-add64 r0, r5
-jlt r0, r3, +0x608
-mov64 r2, 0
-mov64 r9, 12
-and64 r4, 1
-jne r4, 0, +0x328
-add64 r1, r7
-add64 r1, r2
-mov64 r3, 1
-mov64 r2, 1
-ldxdw r4, [r10-1296]
-jle r4, r0, +0x708
-mov64 r2, 0
-ldxdw r4, [r10-1264]
-jgt r4, r1, +0x718
-ldxdw r4, [r10-1264]
-jeq r4, r1, +0x730
-mov64 r2, r3
-and64 r2, 1
-jne r2, 0, +0x740
-ja +0x328
-ldxdw r4, [r10-1264]
-jle r4, r1, +0x6d8
-mov64 r3, 0
-ldxdw r4, [r10-1264]
-jne r4, r1, +0x6e8
-and64 r2, 1
-jeq r2, 0, +0x328
-ldxdw r1, [r10-1304]
-and64 r1, 65535
-jge r1, 10001, +0x35d0
-ldxdw r7, [r8+8]
-ldxdw r1, [r8+32]
-ldxdw r6, [r1+104]
-ldxdw r1, [r10-1312]
-mov64 r2, 0
+ldxdw r3, [r10-2832]
+mov64 r4, 0
+call __multi3
+mov64 r2, r1
 mov64 r3, r6
-mov64 r4, 0
-call __multi3
-mov64 r2, r1
-mov64 r3, r6
-ldxdw r1, [r10-1272]
+ldxdw r1, [r10-2824]
 add64 r3, r1
 mov64 r4, 1
-jlt r3, r6, +0x7d0
+jlt r3, r6, +0x928
 mov64 r4, 0
 mov64 r1, r0
 call __udivti3
-mov64 r9, r0
-jne r1, 0, +0x34d8
+mov64 r8, r0
+jne r1, 0, +0x1fe8
 mov64 r3, 10000
-ldxdw r1, [r10-1304]
+ldxdw r1, [r10-2808]
 sub64 r3, r1
+ldxdw r1, [r10-2848]
+ldxdw r1, [r1+96]
 and64 r3, 65535
-ldxdw r1, [r10-1280]
-ldxdw r1, [r1+96]
 mov64 r2, 0
 mov64 r4, 0
 call __multi3
@@ -263,25 +299,27 @@
 mov64 r3, 10000
 mov64 r4, 0
 call __udivti3
-ldxdw r1, [r7+104]
-mov64 r2, r1
-jlt r1, r9, +0x1b80
-mov64 r2, r9
-jne r1, 0, +0x1b88
-jlt r9, r0, +0x898
-mov64 r9, r0
-mov64 r6, r9
-ldxdw r1, [r10-1312]
+ldxdw r1, [r7+8]
+ldxdw r1, [r1+112]
+mov64 r2, r1
+jlt r1, r8, +0xf00
+mov64 r2, r8
+jne r1, 0, +0xf08
+jlt r8, r0, +0x9f8
+mov64 r8, r0
+mov64 r6, r8
+ldxdw r1, [r10-2832]
 add64 r6, r1
 mov64 r7, 1
-jlt r6, r9, +0x8c8
+jlt r6, r8, +0xa28
 mov64 r7, 0
 mov64 r1, r6
 or64 r1, r7
-jeq r1, 0, +0x3528
-mov64 r1, r9
+jeq r1, 0, +0x2038
+mov64 r9, 0
+mov64 r1, r8
 mov64 r2, 0
-ldxdw r3, [r10-1272]
+ldxdw r3, [r10-2824]
 mov64 r4, 0
 call __multi3
 mov64 r2, r1
@@ -289,1315 +327,558 @@
 mov64 r3, r6
 mov64 r4, r7
 call __udivti3
-jne r1, 0, +0x3580
-ldxdw r1, [r10-1280]
+stxdw [r10-2800], r0
+jne r1, 0, +0x2050
+ldxdw r1, [r10-2848]
 ldxdw r1, [r1+88]
-jlt r1, r0, +0x958
-mov64 r1, r0
-stxdw [r10-1368], r1
-ldxdw r1, [r8+40]
+stxdw [r10-2824], r1
+ldxdw r6, [r10-2840]
+ldxdw r1, [r6+40]
 call solana_account_info::AccountInfo::lamports#aa02
-ldxdw r1, [r8+32]
-ldxdw r7, [r1+104]
-ldxdw r1, [r8+112]
-ldxdw r1, [r1]
-ldxdw r2, [r1+24]
-stxdw [r10-1216], r2
-ldxdw r2, [r1+16]
-stxdw [r10-1224], r2
-ldxdw r2, [r1+8]
-stxdw [r10-1232], r2
-ldxdw r1, [r1]
-stxdw [r10-1240], r1
-ldxdw r2, [r8+64]
-ldxdw r6, [r2+8]
-ldxdw r1, [r6]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xa08
-mov64 r3, 0
-ldxdw r4, [r2]
-stxdw [r6], r1
-jeq r3, 1, +0x3470
-ldxdw r1, [r2+16]
-stxdw [r10-1248], r1
-ldxdw r1, [r1]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xa58
-mov64 r3, 0
-stxdw [r10-1376], r4
-stxdw [r10-2360], r0
-ldxdw r4, [r10-1248]
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-ldxdw r1, [r8+80]
-ldxdw r0, [r1+8]
-ldxdw r3, [r0]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xab8
-mov64 r4, 0
-ldxb r5, [r2+42]
-stxdw [r10-1424], r5
-ldxb r5, [r2+41]
-stxdw [r10-1416], r5
-ldxb r5, [r2+40]
-stxdw [r10-1408], r5
-ldxdw r5, [r2+32]
-stxdw [r10-1400], r5
-ldxdw r2, [r2+24]
-stxdw [r10-1392], r2
+stxdw [r10-2848], r0
+ldxdw r1, [r6+32]
+ldxdw r1, [r1+104]
+stxdw [r10-2832], r1
+ldxdw r2, [r6+64]
+mov64 r7, r10
+add64 r7, -2792
+mov64 r1, r7
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+80]
+mov64 r1, r10
+add64 r1, -2744
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r6+24]
+ldxdw r2, [r1+88]
+mov64 r1, r10
+add64 r1, -2696
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r6+72]
 ldxdw r2, [r1]
-stxdw [r10-1384], r2
-stxdw [r0], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xb58
-mov64 r3, 0
-stxdw [r10-1432], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-stxdw [r10-1440], r6
-ldxdw r2, [r8+24]
-ldxdw r2, [r2+88]
-ldxdw r5, [r2+8]
-ldxdw r3, [r5]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xbb8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1488], r6
-ldxb r6, [r1+41]
-stxdw [r10-1480], r6
-ldxb r6, [r1+40]
-stxdw [r10-1472], r6
-ldxdw r6, [r1+32]
-stxdw [r10-1464], r6
-ldxdw r1, [r1+24]
-stxdw [r10-1456], r1
-ldxdw r1, [r2]
-stxdw [r10-1448], r1
-stxdw [r5], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xc58
-mov64 r3, 0
-stxdw [r10-1496], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-stxdw [r10-2368], r7
-ldxdw r1, [r8+72]
-ldxdw r1, [r1]
-ldxdw r6, [r1+8]
-ldxdw r3, [r6]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xcb8
-mov64 r4, 0
-ldxb r7, [r2+42]
-stxdw [r10-1520], r7
-ldxb r7, [r2+41]
-stxdw [r10-1512], r7
-ldxb r7, [r2+40]
-stxdw [r10-1504], r7
-ldxdw r7, [r2+32]
-stxdw [r10-1528], r7
-ldxdw r2, [r2+24]
-stxdw [r10-1536], r2
-ldxdw r7, [r1]
-stxdw [r6], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xd50
-mov64 r3, 0
-stxdw [r10-1544], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-stxdw [r10-1552], r7
-ldxdw r2, [r8+88]
-ldxdw r2, [r2+32]
-ldxdw r3, [r2+8]
-stxdw [r10-1256], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xdb8
-mov64 r4, 0
-ldxb r7, [r1+42]
-stxdw [r10-1584], r7
-ldxb r7, [r1+41]
-stxdw [r10-1576], r7
-ldxb r7, [r1+40]
-stxdw [r10-1568], r7
-ldxdw r7, [r1+32]
-ldxdw r1, [r1+24]
-stxdw [r10-1592], r1
-ldxdw r1, [r2]
-stxdw [r10-1560], r1
-ldxdw r1, [r10-1256]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0xe58
-mov64 r3, 0
-stxdw [r10-1608], r7
-stxdw [r10-1600], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-stxdw [r10-1616], r6
-ldxdw r1, [r8+32]
-ldxdw r1, [r1+32]
-ldxdw r7, [r1+8]
-ldxdw r3, [r7]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xec0
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1632], r6
-ldxb r6, [r2+41]
-stxdw [r10-1656], r6
-ldxb r6, [r2+40]
-stxdw [r10-1664], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1648], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1640], r2
-ldxdw r2, [r1]
-stxdw [r10-1624], r2
-stxdw [r7], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0xf60
-mov64 r3, 0
-stxdw [r10-1672], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-ldxdw r2, [r8+40]
-ldxdw r3, [r2+8]
-stxdw [r10-1264], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0xfb8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1688], r6
-ldxb r6, [r1+41]
-stxdw [r10-1704], r6
-ldxb r6, [r1+40]
-stxdw [r10-1712], r6
-ldxdw r6, [r1+32]
-stxdw [r10-1696], r6
-ldxdw r6, [r1+24]
-ldxdw r1, [r2]
-stxdw [r10-1680], r1
-ldxdw r1, [r10-1264]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1058
-mov64 r3, 0
-stxdw [r10-1720], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-stxdw [r10-1728], r6
-ldxdw r1, [r8+48]
-ldxdw r3, [r1+8]
-stxdw [r10-1272], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x10b8
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1776], r6
-ldxb r6, [r2+41]
-stxdw [r10-1768], r6
-ldxb r6, [r2+40]
-stxdw [r10-1760], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1752], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1744], r2
-ldxdw r2, [r1]
-stxdw [r10-1736], r2
-ldxdw r2, [r10-1272]
-stxdw [r2], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1160
-mov64 r3, 0
-stxdw [r10-1784], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-ldxdw r2, [r8+56]
-ldxdw r3, [r2+8]
-stxdw [r10-1280], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x11b8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1824], r6
-ldxb r6, [r1+41]
-stxdw [r10-1816], r6
-ldxb r6, [r1+40]
-stxdw [r10-1808], r6
-ldxdw r6, [r1+32]
-stxdw [r10-1800], r6
-ldxdw r1, [r1+24]
-stxdw [r10-1792], r1
-ldxdw r6, [r2]
-ldxdw r1, [r10-1280]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1258
-mov64 r3, 0
-stxdw [r10-1832], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-stxdw [r10-1840], r6
-ldxdw r1, [r8+96]
-ldxdw r3, [r1+8]
-stxdw [r10-1296], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x12b8
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-1888], r6
-ldxb r6, [r2+41]
-stxdw [r10-1880], r6
-ldxb r6, [r2+40]
-stxdw [r10-1872], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1864], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1856], r2
-ldxdw r2, [r1]
-stxdw [r10-1848], r2
-ldxdw r2, [r10-1296]
-stxdw [r2], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1360
-mov64 r3, 0
-stxdw [r10-1896], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-ldxdw r2, [r8+104]
-ldxdw r3, [r2+8]
-stxdw [r10-1312], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x13b8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-1936], r6
-ldxb r6, [r1+41]
-stxdw [r10-1928], r6
-ldxb r6, [r1+40]
-stxdw [r10-1920], r6
-ldxdw r6, [r1+32]
-stxdw [r10-1912], r6
-ldxdw r1, [r1+24]
-stxdw [r10-1904], r1
-ldxdw r6, [r2]
-ldxdw r1, [r10-1312]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1458
-mov64 r3, 0
-stxdw [r10-1944], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-stxdw [r10-1952], r6
-ldxdw r1, [r8+112]
-ldxdw r3, [r1+8]
-stxdw [r10-1328], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x14b8
-mov64 r4, 0
-ldxb r6, [r2+42]
-stxdw [r10-2000], r6
-ldxb r6, [r2+41]
-stxdw [r10-1992], r6
-ldxb r6, [r2+40]
-stxdw [r10-1984], r6
-ldxdw r6, [r2+32]
-stxdw [r10-1976], r6
-ldxdw r2, [r2+24]
-stxdw [r10-1968], r2
-ldxdw r2, [r1]
-stxdw [r10-1960], r2
-ldxdw r2, [r10-1328]
-stxdw [r2], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-jeq r2, 0, +0x1560
-mov64 r3, 0
-stxdw [r10-2008], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-ldxdw r2, [r8+120]
-ldxdw r3, [r2+8]
-stxdw [r10-1336], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x15b8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-2056], r6
-ldxb r6, [r1+41]
-stxdw [r10-2048], r6
-ldxb r6, [r1+40]
-stxdw [r10-2040], r6
-ldxdw r6, [r1+32]
-stxdw [r10-2032], r6
-ldxdw r1, [r1+24]
-stxdw [r10-2024], r1
-ldxdw r1, [r2]
+mov64 r1, r10
+add64 r1, -2648
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r6+88]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2600
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r6+32]
+ldxdw r2, [r1+32]
+mov64 r1, r10
+add64 r1, -2552
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+40]
+mov64 r1, r10
+add64 r1, -2504
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+48]
+mov64 r1, r10
+add64 r1, -2456
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+56]
+mov64 r1, r10
+add64 r1, -2408
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+96]
+mov64 r1, r10
+add64 r1, -2360
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+104]
+mov64 r1, r10
+add64 r1, -2312
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+112]
+mov64 r1, r10
+add64 r1, -2264
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+120]
+mov64 r1, r10
+add64 r1, -2216
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+128]
+mov64 r1, r10
+add64 r1, -2168
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+136]
+mov64 r1, r10
+add64 r1, -2120
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r6+144]
+mov64 r1, r10
+add64 r1, -2072
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r1, [r10-2816]
+ldxb r1, [r1+34]
+stxb [r10-2017], r1
+mov64 r1, r10
+add64 r1, -2017
+stxdw [r10-2000], r1
+lddw r1, .rodata[736f6c2d7661756c]
 stxdw [r10-2016], r1
-ldxdw r1, [r10-1336]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-mov64 r6, r7
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1668
-mov64 r3, 0
-mov64 r7, r5
-stxdw [r10-2064], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-mov64 r5, r0
-ldxdw r1, [r8+128]
-ldxdw r3, [r1+8]
-stxdw [r10-1344], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x16d0
-mov64 r4, 0
-ldxb r0, [r2+42]
-stxdw [r10-2112], r0
-ldxb r0, [r2+41]
-stxdw [r10-2104], r0
-ldxb r0, [r2+40]
-stxdw [r10-2096], r0
-ldxdw r0, [r2+32]
-stxdw [r10-2088], r0
-ldxdw r2, [r2+24]
-stxdw [r10-2080], r2
-ldxdw r2, [r1]
-stxdw [r10-2072], r2
-ldxdw r2, [r10-1344]
-stxdw [r2], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-mov64 r0, r5
-jeq r2, 0, +0x1780
-mov64 r3, 0
-stxdw [r10-2120], r4
-stxdw [r4], r2
-mov64 r5, r7
-jeq r3, 1, +0x3470
-ldxdw r2, [r8+136]
-ldxdw r3, [r2+8]
-stxdw [r10-1352], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-mov64 r7, r6
-jeq r3, 0, +0x17e8
-mov64 r4, 0
-ldxb r6, [r1+42]
-stxdw [r10-2160], r6
-ldxb r6, [r1+41]
-stxdw [r10-2152], r6
-ldxb r6, [r1+40]
-stxdw [r10-2144], r6
-ldxdw r6, [r1+32]
-stxdw [r10-2136], r6
-ldxdw r1, [r1+24]
-stxdw [r10-2128], r1
-ldxdw r6, [r2]
-ldxdw r1, [r10-1352]
-stxdw [r1], r3
-jeq r4, 1, +0x3470
-stxdw [r10-2168], r6
-ldxdw r4, [r2+16]
-ldxdw r1, [r4]
-add64 r1, 1
-mov64 r3, 1
-jeq r1, 0, +0x1890
-mov64 r3, 0
-mov64 r6, r7
-stxdw [r10-2176], r4
-stxdw [r4], r1
-jeq r3, 1, +0x3470
-mov64 r7, r9
-ldxdw r1, [r8+144]
-ldxdw r3, [r1+8]
-stxdw [r10-1360], r3
-ldxdw r3, [r3]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x18f8
-mov64 r4, 0
-ldxb r9, [r2+42]
-stxdw [r10-2224], r9
-ldxb r9, [r2+41]
-stxdw [r10-2216], r9
-ldxb r9, [r2+40]
-stxdw [r10-2208], r9
-ldxdw r9, [r2+32]
-stxdw [r10-2200], r9
-ldxdw r2, [r2+24]
-stxdw [r10-2192], r2
-ldxdw r2, [r1]
-stxdw [r10-2184], r2
-ldxdw r2, [r10-1360]
-stxdw [r2], r3
-jeq r4, 1, +0x3470
-ldxdw r4, [r1+16]
-ldxdw r2, [r4]
-add64 r2, 1
-mov64 r3, 1
-mov64 r9, r7
-jeq r2, 0, +0x19a8
-mov64 r3, 0
-stxdw [r10-2240], r5
-stxdw [r10-2232], r0
-stxdw [r10-2248], r4
-stxdw [r4], r2
-jeq r3, 1, +0x3470
-ldxb r2, [r1+42]
-stxdw [r10-2280], r2
-ldxb r2, [r1+41]
-stxdw [r10-2272], r2
-ldxb r2, [r1+40]
-stxdw [r10-2264], r2
-ldxdw r2, [r1+32]
-stxdw [r10-2256], r2
-ldxdw r7, [r1+24]
-ldxdw r1, [r10-1320]
-ldxb r1, [r1+34]
-mov64 r2, r10
-add64 r2, -305
-stxdw [r10-136], r2
-lddw r2, .rodata[736f6c2d7661756c]
-stxdw [r10-152], r2
-stxb [r10-305], r1
-mov64 r1, r10
-add64 r1, -152
-stxdw [r10-328], r1
-ldxdw r1, [r10-1240]
-stxdw [r10-72], r1
-ldxdw r1, [r10-1232]
-stxdw [r10-64], r1
-ldxdw r1, [r10-1224]
-stxdw [r10-56], r1
-ldxdw r1, [r10-1216]
-stxdw [r10-48], r1
-stdw [r10-128], 1
-stdw [r10-144], 9
-stdw [r10-320], 2
+stdw [r10-1992], 1
+stdw [r10-2008], 9
+mov64 r1, r10
+add64 r1, -2016
+stxdw [r10-1984], r1
+stdw [r10-1976], 2
+mov64 r1, r10
+add64 r1, -1104
+mov64 r2, r7
+mov64 r3, 768
+call memcpy
+lddw r1, 0xcf5a6693f6e05601
+stxdw [r10-320], r1
+lddw r1, 0xaa5b17bf6815db44
+stxdw [r10-312], r1
+lddw r1, 0x3bffd2f597cb8951
+stxdw [r10-304], r1
+lddw r1, 0xb0186dfdb62b5d65
+stxdw [r10-296], r1
+mov64 r1, r10
+add64 r1, -1984
+stxdw [r10-336], r1
+stdw [r10-1112], 0
+stdw [r10-1120], 8
+stdw [r10-1128], 0
+stdw [r10-328], 1
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
-ldxdw r3, [r1]
-mov64 r2, 0
-mov64 r4, 1
-mov64 r1, r3
-add64 r1, -96
-jgt r1, r3, +0x1ba0
-mov64 r4, 0
-jeq r4, 0, +0x1ba8
-lddw r1, 0x300007fa0
-jeq r3, 0, +0x1bc8
-and64 r2, -8
-mov64 r1, r2
-lddw r2, 0x300000008
-jge r1, r2, +0x1be0
+ldxdw r1, [r1]
+mov64 r3, 1
+mov64 r2, r1
+add64 r2, -96
+jgt r2, r1, +0xf20
+mov64 r3, 0
+jeq r3, 0, +0xf28
+lddw r7, 0x300007fa0
+jeq r1, 0, +0xf48
+and64 r9, -8
+mov64 r7, r9
+lddw r1, 0x300000008
+jge r7, r1, +0xf60
 mov64 r1, 8
 mov64 r2, 96
 call alloc::alloc::handle_alloc_error#decd
-jeq r1, 0, +0x888
+jeq r1, 0, +0x9e8
+mov64 r8, r2
+jge r8, r0, +0x9f0
+ja +0x9f8
+jne r3, 0, +0xea8
 mov64 r9, r2
-jge r9, r0, +0x890
-ja +0x898
-jne r4, 0, +0x1b28
-mov64 r2, r1
-lddw r1, 0x300007fa0
-jne r3, 0, +0x1b40
-lddw r2, 0x300000008
-jlt r1, r2, +0x1b68
-lddw r2, 0x300000000
-stxdw [r2], r1
-ldxdw r5, [r8+168]
-ldxdw r2, [r5+8]
-ldxdw r3, [r2]
-add64 r3, 1
-mov64 r4, 1
-jeq r3, 0, +0x1c30
-mov64 r4, 0
-ldxdw r0, [r5]
-stxdw [r10-1320], r0
-stxdw [r2], r3
-and64 r4, 1
-jne r4, 0, +0x3470
-ldxdw r4, [r5+16]
-ldxdw r3, [r4]
-add64 r3, 1
-mov64 r0, 1
-jeq r3, 0, +0x1c88
-mov64 r0, 0
-stxdw [r10-2296], r7
-stxdw [r10-2288], r6
-stxdw [r4], r3
-jeq r0, 1, +0x3470
-ldxdw r6, [r8+176]
-ldxdw r0, [r6+8]
-ldxdw r7, [r0]
-add64 r7, 1
-mov64 r8, 1
-jeq r7, 0, +0x1ce0
-mov64 r8, 0
-ldxb r3, [r5+42]
-stxdw [r10-2328], r3
-ldxb r3, [r5+41]
-stxdw [r10-2320], r3
-ldxb r3, [r5+40]
-stxdw [r10-2312], r3
-ldxdw r3, [r5+32]
-stxdw [r10-2304], r3
-ldxdw r3, [r5+24]
-ldxdw r5, [r6]
-stxdw [r10-2336], r5
-stxdw [r0], r7
-jeq r8, 1, +0x3470
-stxdw [r10-2352], r3
-ldxdw r5, [r6+16]
-ldxdw r7, [r5]
-add64 r7, 1
-mov64 r8, 1
-jeq r7, 0, +0x1d80
-mov64 r8, 0
-mov64 r3, r2
-stxdw [r10-2344], r9
-stxdw [r5], r7
-jeq r8, 1, +0x3470
-ldxdw r7, [r6+24]
-ldxdw r8, [r6+32]
-ldxb r9, [r6+40]
-ldxb r2, [r6+41]
-ldxb r6, [r6+42]
-stxb [r1+90], r6
-stxb [r1+89], r2
-stxb [r1+88], r9
-stxdw [r1+80], r8
-stxdw [r1+72], r7
-stxdw [r1+64], r5
-stxdw [r1+56], r0
-ldxdw r2, [r10-2336]
-stxdw [r1+48], r2
-ldxdw r2, [r10-2328]
-stxb [r1+42], r2
-ldxdw r2, [r10-2320]
-stxb [r1+41], r2
-ldxdw r2, [r10-2312]
-stxb [r1+40], r2
-ldxdw r2, [r10-2304]
-stxdw [r1+32], r2
-ldxdw r2, [r10-2352]
-stxdw [r1+24], r2
-stxdw [r1+16], r4
-stxdw [r1+8], r3
-ldxdw r2, [r10-1320]
-stxdw [r1], r2
-ldxdw r2, [r10-1424]
-stxb [r10-1174], r2
-ldxdw r2, [r10-1416]
-stxb [r10-1175], r2
-ldxdw r2, [r10-1408]
-stxb [r10-1176], r2
-ldxdw r2, [r10-1400]
-stxdw [r10-1184], r2
-ldxdw r2, [r10-1392]
-stxdw [r10-1192], r2
-ldxdw r2, [r10-1248]
-stxdw [r10-1200], r2
-ldxdw r2, [r10-1440]
-stxdw [r10-1208], r2
-ldxdw r2, [r10-1376]
-stxdw [r10-1216], r2
-stxdw [r10-1232], r1
-ldxdw r1, [r10-1488]
-stxb [r10-1126], r1
-ldxdw r1, [r10-1480]
-stxb [r10-1127], r1
-ldxdw r1, [r10-1472]
-stxb [r10-1128], r1
-ldxdw r1, [r10-1464]
-stxdw [r10-1136], r1
-ldxdw r1, [r10-1456]
-stxdw [r10-1144], r1
-ldxdw r1, [r10-1432]
-stxdw [r10-1152], r1
-ldxdw r1, [r10-2232]
-stxdw [r10-1160], r1
-ldxdw r1, [r10-1384]
-stxdw [r10-1168], r1
-ldxw r1, [r10-304]
-stxw [r10-1173], r1
-ldxb r1, [r10-300]
-stxb [r10-1169], r1
-stdw [r10-1224], 2
-stdw [r10-1240], 2
-ldxdw r1, [r10-1448]
+lddw r7, 0x300007fa0
+jne r1, 0, +0xec0
+lddw r1, 0x300000008
+jlt r7, r1, +0xee8
+lddw r1, 0x300000000
+stxdw [r1], r7
+ldxdw r9, [r10-2840]
+ldxdw r2, [r9+168]
+mov64 r6, r10
+add64 r6, -56
+mov64 r1, r6
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+ldxdw r2, [r9+176]
+mov64 r9, r10
+add64 r9, -136
+mov64 r1, r9
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r7
+mov64 r2, r6
+mov64 r3, 48
+call memcpy
+mov64 r1, r7
+add64 r1, 48
+mov64 r2, r9
+mov64 r3, 48
+call memcpy
+mov64 r6, r10
+add64 r6, -1128
+mov64 r1, r6
+call core::ptr::drop_in_place<alloc::vec::Vec<solana_account_info::AccountInfo>>#d146
+stxdw [r10-1120], r7
+stdw [r10-1112], 2
+stdw [r10-1128], 2
+mov64 r7, r10
+add64 r7, -1968
+mov64 r1, r7
+mov64 r2, r6
+mov64 r3, 840
+call memcpy
+ldxdw r1, [r10-2800]
+ldxdw r2, [r10-2824]
+jlt r2, r1, +0x10a0
+mov64 r2, r1
+mov64 r1, r7
+mov64 r3, r8
+mov64 r4, 0
+call mayhem_program::pump::cpi::buy#e82e
+mov64 r6, r0
+mov64 r2, r1
+mov64 r1, r6
+call core::ptr::drop_in_place<core::result::Result<(),anchor_lang::error::Error>>#64a0
+jne r6, 2, +0x1378
+ldxdw r6, [r10-2840]
+ldxdw r1, [r6+32]
+call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
+jne r0, 2, +0x430
+ldxdw r1, [r6+72]
+call anchor_lang::accounts::account::Account<T>::reload#fa41
+jne r0, 2, +0x430
+ldxdw r1, [r6+40]
+call solana_account_info::AccountInfo::lamports#aa02
+ldxdw r8, [r10-2848]
+jlt r8, r0, +0x20b8
+ldxdw r1, [r6+32]
+ldxdw r7, [r1+104]
+ldxdw r1, [r10-2832]
+jlt r7, r1, +0x20d0
+sub64 r8, r0
+ldxdw r1, [r6+8]
+ldxdw r3, [r1+72]
+mov64 r2, r3
+add64 r2, r8
+mov64 r5, 1
+jlt r2, r3, +0x11a0
+mov64 r5, 0
+ldxdw r4, [r1+80]
+mov64 r3, r4
+add64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+xor64 r4, -1
+and64 r4, r5
+jslt r4, 0, +0x1368
+stxdw [r1+72], r2
+stxdw [r1+80], r3
+ldxdw r1, [r6+16]
+ldxdw r3, [r1+40]
+mov64 r2, r3
+add64 r2, r8
+mov64 r5, 1
+jlt r2, r3, +0x1228
+mov64 r5, 0
+ldxdw r4, [r1+48]
+mov64 r3, r4
+add64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+xor64 r4, -1
+and64 r4, r5
+jsle r4, -1, +0x1368
+ldxdw r4, [r10-2832]
+sub64 r7, r4
+stxdw [r1+40], r2
+stxdw [r1+48], r3
+ldxdw r1, [r6+16]
+ldxdw r2, [r1+56]
+mov64 r5, 1
+jlt r2, r7, +0x12b0
+mov64 r5, 0
+ldxdw r4, [r1+64]
+mov64 r3, r4
+sub64 r3, r5
+mov64 r5, r4
+xor64 r5, r3
+and64 r4, r5
+jsle r4, -1, +0x1368
+sub64 r2, r7
+stxdw [r1+56], r2
+stxdw [r1+64], r3
+ldxdw r1, [r6+8]
+ldxdw r4, [r1+80]
+mov64 r2, 1
+mov64 r5, 1
+jsgt r4, 0, +0x1408
+mov64 r5, 0
+ldxdw r3, [r1+72]
+jeq r3, 0, +0x1418
+jeq r4, 0, +0x1428
+mov64 r2, r5
+and64 r2, 1
+jne r2, 0, +0x1438
+ja +0x1448
+call mayhem_program::instructions::pump_interface::buy::math#52f7
+ja +0x430
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[08000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 8
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
 stxdw [r10-1120], r1
-ldxdw r1, [r10-2240]
-stxdw [r10-1112], r1
-ldxdw r1, [r10-1496]
-stxdw [r10-1104], r1
-ldxdw r1, [r10-1536]
-stxdw [r10-1096], r1
-ldxdw r1, [r10-1528]
-stxdw [r10-1088], r1
-ldxdw r1, [r10-1520]
-stxb [r10-1078], r1
-ldxdw r1, [r10-1512]
-stxb [r10-1079], r1
-ldxdw r1, [r10-1504]
-stxb [r10-1080], r1
-ldxdw r1, [r10-1552]
-stxdw [r10-1072], r1
-ldxdw r1, [r10-1616]
-stxdw [r10-1064], r1
-ldxdw r1, [r10-1544]
-stxdw [r10-1056], r1
-ldxdw r1, [r10-1592]
-stxdw [r10-1048], r1
-ldxdw r1, [r10-1608]
-stxdw [r10-1040], r1
-ldxdw r1, [r10-1584]
-stxb [r10-1030], r1
-ldxdw r1, [r10-1576]
-stxb [r10-1031], r1
-ldxdw r1, [r10-1568]
-stxb [r10-1032], r1
-ldxb r1, [r10-20]
-stxb [r10-1121], r1
-ldxw r1, [r10-24]
-stxw [r10-1125], r1
-ldxw r1, [r10-333]
-stxw [r10-1077], r1
-ldxb r1, [r10-329]
-stxb [r10-1073], r1
-ldxdw r1, [r10-1664]
-stxb [r10-984], r1
-ldxdw r1, [r10-1656]
-stxb [r10-983], r1
-ldxdw r1, [r10-1632]
-stxb [r10-982], r1
-ldxdw r1, [r10-1648]
-stxdw [r10-992], r1
-ldxdw r1, [r10-1640]
-stxdw [r10-1000], r1
-ldxdw r1, [r10-1600]
-stxdw [r10-1008], r1
-ldxdw r1, [r10-1256]
-stxdw [r10-1016], r1
-ldxdw r1, [r10-1560]
-stxdw [r10-1024], r1
-ldxdw r1, [r10-1712]
-stxb [r10-936], r1
-ldxdw r1, [r10-1704]
-stxb [r10-935], r1
-ldxdw r1, [r10-1688]
-stxb [r10-934], r1
-ldxdw r1, [r10-1696]
-stxdw [r10-944], r1
-ldxdw r1, [r10-1728]
-stxdw [r10-952], r1
-ldxdw r1, [r10-1672]
-stxdw [r10-960], r1
-ldxdw r1, [r10-2288]
-stxdw [r10-968], r1
-ldxdw r1, [r10-1624]
-stxdw [r10-976], r1
-ldxb r1, [r10-334]
-stxb [r10-1025], r1
-ldxw r1, [r10-338]
-stxw [r10-1029], r1
-ldxw r1, [r10-343]
-stxw [r10-981], r1
-ldxb r1, [r10-339]
-stxb [r10-977], r1
-ldxb r1, [r10-344]
-stxb [r10-929], r1
-ldxw r1, [r10-348]
-stxw [r10-933], r1
-ldxdw r1, [r10-1776]
-stxb [r10-886], r1
-ldxdw r1, [r10-1768]
-stxb [r10-887], r1
-ldxdw r1, [r10-1760]
-stxb [r10-888], r1
-ldxdw r1, [r10-1752]
-stxdw [r10-896], r1
-ldxdw r1, [r10-1744]
-stxdw [r10-904], r1
-ldxdw r1, [r10-1720]
-stxdw [r10-912], r1
-ldxdw r1, [r10-1264]
-stxdw [r10-920], r1
-ldxdw r1, [r10-1680]
-stxdw [r10-928], r1
-ldxb r1, [r10-349]
-stxb [r10-881], r1
-ldxw r1, [r10-353]
-stxw [r10-885], r1
-ldxdw r1, [r10-1824]
-stxb [r10-838], r1
-ldxdw r1, [r10-1816]
-stxb [r10-839], r1
-ldxdw r1, [r10-1808]
-stxb [r10-840], r1
-ldxdw r1, [r10-1800]
-stxdw [r10-848], r1
-ldxdw r1, [r10-1792]
-stxdw [r10-856], r1
-ldxdw r1, [r10-1784]
-stxdw [r10-864], r1
-ldxdw r1, [r10-1272]
-stxdw [r10-872], r1
-ldxdw r1, [r10-1736]
-stxdw [r10-880], r1
-ldxb r1, [r10-354]
-stxb [r10-833], r1
-ldxw r1, [r10-358]
-stxw [r10-837], r1
-ldxdw r1, [r10-1888]
-stxb [r10-790], r1
-ldxdw r1, [r10-1880]
-stxb [r10-791], r1
-ldxdw r1, [r10-1872]
-stxb [r10-792], r1
-ldxdw r1, [r10-1864]
-stxdw [r10-800], r1
-ldxdw r1, [r10-1856]
-stxdw [r10-808], r1
-ldxdw r1, [r10-1832]
-stxdw [r10-816], r1
-ldxdw r1, [r10-1280]
-stxdw [r10-824], r1
-ldxdw r1, [r10-1840]
-stxdw [r10-832], r1
-ldxb r1, [r10-359]
-stxb [r10-785], r1
-ldxw r1, [r10-363]
-stxw [r10-789], r1
-ldxdw r1, [r10-1936]
-stxb [r10-742], r1
-ldxdw r1, [r10-1928]
-stxb [r10-743], r1
-ldxdw r1, [r10-1920]
-stxb [r10-744], r1
-ldxdw r1, [r10-1912]
-stxdw [r10-752], r1
-ldxdw r1, [r10-1904]
-stxdw [r10-760], r1
-ldxdw r1, [r10-1896]
-stxdw [r10-768], r1
-ldxdw r1, [r10-1296]
-stxdw [r10-776], r1
-ldxdw r1, [r10-1848]
-stxdw [r10-784], r1
-ldxb r1, [r10-364]
-stxb [r10-737], r1
-ldxw r1, [r10-368]
-stxw [r10-741], r1
-ldxdw r1, [r10-2000]
-stxb [r10-694], r1
-ldxdw r1, [r10-1992]
-stxb [r10-695], r1
-ldxdw r1, [r10-1984]
-stxb [r10-696], r1
-ldxdw r1, [r10-1976]
-stxdw [r10-704], r1
-ldxdw r1, [r10-1968]
-stxdw [r10-712], r1
-ldxdw r1, [r10-1944]
-stxdw [r10-720], r1
-ldxdw r1, [r10-1312]
-stxdw [r10-728], r1
-ldxdw r1, [r10-1952]
-stxdw [r10-736], r1
-ldxb r1, [r10-369]
-stxb [r10-689], r1
-ldxw r1, [r10-373]
-stxw [r10-693], r1
-ldxdw r1, [r10-2056]
-stxb [r10-646], r1
-ldxdw r1, [r10-2048]
-stxb [r10-647], r1
-ldxdw r1, [r10-2040]
-stxb [r10-648], r1
-ldxdw r1, [r10-2032]
-stxdw [r10-656], r1
-ldxdw r1, [r10-2024]
-stxdw [r10-664], r1
-ldxdw r1, [r10-2008]
-stxdw [r10-672], r1
-ldxdw r1, [r10-1328]
-stxdw [r10-680], r1
-ldxdw r1, [r10-1960]
-stxdw [r10-688], r1
-ldxb r1, [r10-374]
-stxb [r10-641], r1
-ldxw r1, [r10-378]
-stxw [r10-645], r1
-ldxdw r1, [r10-2112]
-stxb [r10-598], r1
-ldxdw r1, [r10-2104]
-stxb [r10-599], r1
-ldxdw r1, [r10-2096]
-stxb [r10-600], r1
-ldxdw r1, [r10-2088]
-stxdw [r10-608], r1
-ldxdw r1, [r10-2080]
-stxdw [r10-616], r1
-ldxdw r1, [r10-2064]
-stxdw [r10-624], r1
-ldxdw r1, [r10-1336]
-stxdw [r10-632], r1
-ldxdw r1, [r10-2016]
-stxdw [r10-640], r1
-ldxb r1, [r10-379]
-stxb [r10-593], r1
-ldxw r1, [r10-383]
-stxw [r10-597], r1
-ldxdw r1, [r10-2160]
-stxb [r10-550], r1
-ldxdw r1, [r10-2152]
-stxb [r10-551], r1
-ldxdw r1, [r10-2144]
-stxb [r10-552], r1
-ldxdw r1, [r10-2136]
-stxdw [r10-560], r1
-ldxdw r1, [r10-2128]
-stxdw [r10-568], r1
-ldxdw r1, [r10-2120]
-stxdw [r10-576], r1
-ldxdw r1, [r10-1344]
-stxdw [r10-584], r1
-ldxdw r1, [r10-2072]
-stxdw [r10-592], r1
-ldxb r1, [r10-384]
-stxb [r10-545], r1
-ldxw r1, [r10-388]
-stxw [r10-549], r1
-ldxdw r1, [r10-2224]
-stxb [r10-502], r1
-ldxdw r1, [r10-2216]
-stxb [r10-503], r1
-ldxdw r1, [r10-2208]
-stxb [r10-504], r1
-ldxdw r1, [r10-2200]
-stxdw [r10-512], r1
-ldxdw r1, [r10-2192]
-stxdw [r10-520], r1
-ldxdw r1, [r10-2176]
-stxdw [r10-528], r1
-ldxdw r1, [r10-1352]
-stxdw [r10-536], r1
-ldxdw r1, [r10-2168]
-stxdw [r10-544], r1
-ldxb r1, [r10-389]
-stxb [r10-497], r1
-ldxw r1, [r10-393]
-stxw [r10-501], r1
-ldxdw r1, [r10-2280]
-stxb [r10-454], r1
-ldxdw r1, [r10-2272]
-stxb [r10-455], r1
-ldxdw r1, [r10-2264]
-stxb [r10-456], r1
-ldxdw r1, [r10-2256]
-stxdw [r10-464], r1
-ldxdw r1, [r10-2296]
-stxdw [r10-472], r1
-ldxdw r1, [r10-2248]
-stxdw [r10-480], r1
-ldxdw r1, [r10-1360]
-stxdw [r10-488], r1
-ldxdw r1, [r10-2184]
-stxdw [r10-496], r1
-ldxb r1, [r10-394]
-stxb [r10-449], r1
-ldxw r1, [r10-398]
-stxw [r10-453], r1
-mov64 r1, r10
-add64 r1, -328
-stxdw [r10-448], r1
-stdw [r10-440], 1
-ldxdw r1, [r10-48]
-stxdw [r10-408], r1
-ldxdw r1, [r10-56]
-stxdw [r10-416], r1
-ldxdw r1, [r10-64]
-stxdw [r10-424], r1
-ldxdw r1, [r10-72]
-stxdw [r10-432], r1
-mov64 r1, r10
-add64 r1, -1240
-ldxdw r2, [r10-1368]
-ldxdw r3, [r10-2344]
-mov64 r4, 0
-call mayhem_program::pump::cpi::buy#e82e
-mov64 r7, r0
-mov64 r6, r1
-jne r7, 2, +0x2950
-ldxdw r8, [r10-1288]
-ja +0x2af8
-mov64 r1, r10
-add64 r1, -24
-lddw r2, .rodata[08000000]
-call mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError::name#fdf6
-stdw [r10-288], 0
-stdw [r10-296], 1
-stdw [r10-304], 0
-lddw r1, table[0:core::ptr::drop_in_place<alloc::string::String>#91b5, 3:<alloc::string::String as core::fmt::Write>::write_str#ca2d, 4:<alloc::string::String as core::fmt::Write>::write_char#6537, 5:core::fmt::Write::write_fmt#a290]
-stxdw [r10-64], r1
-mov64 r1, r10
-add64 r1, -304
-stxdw [r10-72], r1
-lddw r1, 0xe0000020
-stxdw [r10-56], r1
-mov64 r2, r10
-add64 r2, -72
-lddw r1, .rodata[08000000]
-call <mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Display>::fmt#0036
-ldxdw r8, [r10-1288]
-jne r0, 0, +0x3488
-ldxdw r1, [r10-288]
-stxdw [r10-1168], r1
-ldxdw r1, [r10-296]
-stxdw [r10-1176], r1
-ldxdw r1, [r10-304]
-stxdw [r10-1184], r1
-ldxdw r1, [r10-24]
-stxdw [r10-1208], r1
-ldxdw r1, [r10-16]
-stxdw [r10-1200], r1
-ldxdw r1, [r10-8]
-stxdw [r10-1192], r1
-lddw r1, .rodata[70726f6772616d73]
-stxdw [r10-1232], r1
-stw [r10-1088], 6008
-stb [r10-1160], 2
-stw [r10-1216], 313
-stdw [r10-1224], 58
-stdw [r10-1240], 0
-mov64 r1, r10
-add64 r1, -1240
-call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
-jeq r7, 0, +0x2ae8
-ldxdw r2, [r6]
-ldxb r2, [r6+40]
-ja +0x2af0
-ldxdw r2, [r6]
-jne r0, 2, +0x480
-ldxdw r1, [r8+32]
-call anchor_lang::accounts::interface_account::InterfaceAccount<T>::reload#d655
-jne r0, 2, +0x480
-ldxdw r1, [r8+72]
-call anchor_lang::accounts::account::Account<T>::reload#fa41
-jne r0, 2, +0x480
-ldxdw r1, [r8+40]
-call solana_account_info::AccountInfo::lamports#aa02
-ldxdw r3, [r10-2360]
-jlt r3, r0, +0x35e8
-ldxdw r1, [r8+32]
-ldxdw r6, [r1+104]
-ldxdw r1, [r10-2368]
-jlt r6, r1, +0x3600
-sub64 r3, r0
-sub64 r6, r1
-ldxdw r2, [r8+16]
-ldxdw r1, [r8+8]
-stdw [r10-1216], 0
-stdw [r10-1224], 0
-stdw [r10-1232], 0
-stdw [r10-1240], 0
-add64 r1, 8
-add64 r2, 8
-mov64 r5, r10
-add64 r5, -1240
-mov64 r7, r3
-mov64 r4, r6
-call orig_fn_2c478
-jne r0, 2, +0x480
-ldxdw r1, [r8+40]
+stb [r10-1048], 2
+stw [r10-1104], 88
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6008
+ja +0x418
+ldxdw r3, [r1+72]
+jne r3, 0, +0x1340
+mov64 r2, 0
+jne r4, 0, +0x1348
+and64 r2, 1
+jeq r2, 0, +0x1448
+ldxdw r2, [r1+128]
+jle r2, r3, +0x1690
+ldxdw r3, [r6+16]
+ldxdw r2, [r3+40]
+ldxdw r4, [r3+48]
+mov64 r3, 1
+mov64 r5, 1
+jsgt r4, 0, +0x14b0
+mov64 r5, 0
+jeq r2, 0, +0x14b8
+jeq r4, 0, +0x14c8
+mov64 r3, r5
+and64 r3, 1
+jne r3, 0, +0x14d8
+ja +0x14e8
+jne r2, 0, +0x1488
+mov64 r3, 0
+jne r4, 0, +0x1490
+and64 r3, 1
+jeq r3, 0, +0x14e8
+ldxdw r1, [r1+136]
+jle r1, r2, +0x1de8
+stxdw [r10-2848], r8
+ldxdw r1, [r6+40]
 ldxdw r2, [r1]
 ldxdw r3, [r2+24]
+stxdw [r10-264], r3
+ldxdw r3, [r2+16]
+stxdw [r10-272], r3
+ldxdw r3, [r2+8]
 stxdw [r10-280], r3
-ldxdw r3, [r2+16]
-stxdw [r10-288], r3
-ldxdw r3, [r2+8]
-stxdw [r10-296], r3
 ldxdw r2, [r2]
-stxdw [r10-304], r2
-ldxdw r2, [r8+24]
+stxdw [r10-288], r2
+ldxdw r2, [r6+24]
 ldxdw r2, [r2+88]
 ldxdw r2, [r2]
 ldxdw r3, [r2+24]
+stxdw [r10-32], r3
+ldxdw r3, [r2+16]
+stxdw [r10-40], r3
+ldxdw r3, [r2+8]
 stxdw [r10-48], r3
-ldxdw r3, [r2+16]
-stxdw [r10-56], r3
-ldxdw r3, [r2+8]
-stxdw [r10-64], r3
 ldxdw r2, [r2]
-stxdw [r10-72], r2
+stxdw [r10-56], r2
 call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-1248], r0
-ldxdw r1, [r8+32]
-ldxdw r1, [r1+104]
-stxdw [r10-1256], r1
-ldxdw r1, [r8+16]
+stxdw [r10-2856], r0
+ldxdw r1, [r6+16]
 ldxdw r2, [r1+48]
-stxdw [r10-1296], r2
+stxdw [r10-2816], r2
 ldxdw r2, [r1+40]
-stxdw [r10-1288], r2
+stxdw [r10-2800], r2
 ldxdw r2, [r1+64]
-stxdw [r10-1312], r2
-ldxdw r9, [r1+56]
+stxdw [r10-2872], r2
+ldxdw r1, [r1+56]
+stxdw [r10-2864], r1
+ldxdw r1, [r6+8]
+ldxdw r2, [r1+80]
+stxdw [r10-2832], r2
+ldxdw r1, [r1+72]
+stxdw [r10-2824], r1
+ldxdw r1, [r6+32]
+ldxdw r9, [r1+104]
+ldxdw r1, [r6+72]
+ldxdw r2, [r1+80]
+stxdw [r10-2880], r2
+ldxdw r8, [r1+72]
+mov64 r1, r10
+add64 r1, -136
+call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
+ldxw r1, [r10-136]
+jeq r1, 0, +0x1758
+ldxw r2, [r10-128]
+ldxw r1, [r10-132]
+call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
+ja +0x430
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[06000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 6
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 96
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6006
+mov64 r1, r10
+add64 r1, -1128
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+mov64 r2, r1
+ldxdw r1, [r6+8]
+ldxdw r4, [r1+72]
+ldxdw r3, [r1+128]
+ja +0x1eb0
+ldxdw r1, [r10-96]
+stxdw [r10-2888], r1
+ldxdw r1, [r6+16]
 ldxdw r1, [r1+80]
-stxdw [r10-1272], r1
-ldxdw r1, [r8+72]
-ldxdw r2, [r1+80]
-stxdw [r10-1264], r2
-ldxdw r1, [r1+72]
-stxdw [r10-1280], r1
-mov64 r1, r10
-add64 r1, -152
-call solana_sysvar::clock::<impl solana_sysvar::Sysvar for solana_clock::Clock>::get#06ec
-ldxw r1, [r10-152]
-jne r1, 1, +0x2d78
-ldxw r2, [r10-144]
-ldxw r1, [r10-148]
-call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
-ja +0x480
-ldxdw r1, [r10-112]
-stxdw [r10-1320], r1
-ldxdw r1, [r8+72]
+stxdw [r10-2896], r1
+ldxdw r1, [r6+72]
 ldxdw r2, [r1+96]
-stxdw [r10-1344], r2
+stxdw [r10-2904], r2
 ldxdw r1, [r1+88]
-stxdw [r10-1352], r1
-ldxdw r1, [r8+8]
-ldxw r2, [r1+248]
-stxdw [r10-1360], r2
+stxdw [r10-2912], r1
+ldxdw r1, [r6+8]
+ldxw r1, [r1+224]
+stxdw [r10-2936], r1
+ldxdw r1, [r6+40]
+call solana_account_info::AccountInfo::lamports#aa02
+ldxdw r1, [r6+8]
 ldxdw r2, [r1+72]
-stxdw [r10-1368], r2
+stxdw [r10-2920], r2
 ldxdw r1, [r1+80]
-stxdw [r10-1376], r1
-ldxdw r1, [r8+40]
-call solana_account_info::AccountInfo::lamports#aa02
-stxdw [r10-1328], r0
-ldxdw r1, [r8+72]
-ldxdw r2, [r1+80]
-stxdw [r10-1336], r2
-ldxdw r2, [r1+96]
-ldxdw r1, [r8+16]
-ldxdw r3, [r1+40]
-ldxdw r4, [r1+48]
-ldxdw r1, [r8+8]
-ldxdw r5, [r1+72]
-ldxdw r1, [r1+80]
-ldxdw r0, [r10-296]
-stxdw [r10-1232], r0
-ldxdw r0, [r10-288]
-stxdw [r10-1224], r0
-ldxdw r0, [r10-280]
-stxdw [r10-1216], r0
-ldxdw r0, [r10-304]
-stxdw [r10-1240], r0
-stdw [r10-960], 0
-stdw [r10-968], 0
-stdw [r10-976], 0
-stdw [r10-984], 0
-stb [r10-946], 0
-ldxdw r0, [r10-72]
-stxdw [r10-1208], r0
-ldxdw r0, [r10-64]
-stxdw [r10-1200], r0
-ldxdw r0, [r10-56]
-stxdw [r10-1192], r0
-ldxdw r0, [r10-48]
-stxdw [r10-1184], r0
-ldxdw r0, [r10-1376]
-stxdw [r10-1088], r0
-ldxdw r0, [r10-1368]
-stxdw [r10-1096], r0
-ldxdw r0, [r10-1312]
-stxdw [r10-1120], r0
-stxdw [r10-1128], r9
-ldxdw r0, [r10-1296]
-stxdw [r10-1136], r0
-ldxdw r0, [r10-1288]
-stxdw [r10-1144], r0
-ldxdw r0, [r10-1304]
-stxh [r10-948], r0
-ldxdw r0, [r10-1360]
-stxw [r10-952], r0
-ldxdw r0, [r10-1352]
-stxdw [r10-1056], r0
-ldxdw r0, [r10-1344]
-stxdw [r10-1064], r0
-ldxdw r0, [r10-1272]
-stxdw [r10-1072], r0
-ldxdw r0, [r10-1320]
-stxdw [r10-1080], r0
-ldxdw r0, [r10-1280]
-stxdw [r10-1104], r0
-ldxdw r0, [r10-1264]
-stxdw [r10-1112], r0
-ldxdw r0, [r10-1256]
-stxdw [r10-1152], r0
-ldxdw r0, [r10-1248]
-stxdw [r10-1160], r0
-stxdw [r10-1168], r6
-stxdw [r10-1176], r7
-stxdw [r10-1000], r1
-stxdw [r10-1008], r5
-stxdw [r10-1024], r4
-stxdw [r10-1032], r3
+stxdw [r10-2928], r1
+ldxdw r1, [r6+72]
+ldxdw r3, [r1+96]
+ldxdw r4, [r1+80]
+ldxdw r1, [r6+16]
+ldxdw r5, [r1+40]
+ldxdw r1, [r1+48]
+ldxdw r2, [r10-280]
+stxdw [r10-1120], r2
+ldxdw r2, [r10-272]
+stxdw [r10-1112], r2
+ldxdw r2, [r10-264]
+stxdw [r10-1104], r2
+ldxdw r2, [r10-288]
+stxdw [r10-1128], r2
+stdw [r10-1040], 0
+stdw [r10-1048], 0
+stdw [r10-1056], 0
+stdw [r10-1064], 0
+stb [r10-834], 0
+ldxdw r2, [r10-56]
+stxdw [r10-1096], r2
+ldxdw r2, [r10-48]
+stxdw [r10-1088], r2
+ldxdw r2, [r10-40]
+stxdw [r10-1080], r2
+ldxdw r2, [r10-32]
+stxdw [r10-1072], r2
+stxdw [r10-928], r9
+ldxdw r2, [r10-2856]
+stxdw [r10-936], r2
+stxdw [r10-944], r7
+ldxdw r7, [r10-2848]
+stxdw [r10-952], r7
+ldxdw r2, [r10-2872]
+stxdw [r10-1008], r2
+ldxdw r2, [r10-2864]
+stxdw [r10-1016], r2
+ldxdw r2, [r10-2816]
+stxdw [r10-1024], r2
+ldxdw r2, [r10-2800]
+stxdw [r10-1032], r2
+stxdw [r10-912], r8
+ldxdw r2, [r10-2880]
+stxdw [r10-920], r2
+ldxdw r2, [r10-2832]
 stxdw [r10-992], r2
-ldxdw r1, [r10-1336]
-stxdw [r10-1016], r1
-ldxdw r1, [r10-1328]
-stxdw [r10-1040], r1
-stxdw [r10-1048], r7
-mov64 r1, r10
-add64 r1, -152
-mov64 r6, r10
-add64 r6, -1240
-mov64 r2, r6
+ldxdw r2, [r10-2824]
+stxdw [r10-1000], r2
+ldxdw r2, [r10-2808]
+stxh [r10-836], r2
+ldxdw r2, [r10-2936]
+stxw [r10-840], r2
+ldxdw r2, [r10-2912]
+stxdw [r10-880], r2
+ldxdw r2, [r10-2904]
+stxdw [r10-888], r2
+ldxdw r2, [r10-2896]
+stxdw [r10-896], r2
+ldxdw r2, [r10-2888]
+stxdw [r10-904], r2
+stxdw [r10-864], r0
+stxdw [r10-872], r7
+stxdw [r10-976], r1
+stxdw [r10-984], r5
+stxdw [r10-856], r4
+ldxdw r1, [r10-2928]
+stxdw [r10-960], r1
+ldxdw r1, [r10-2920]
+stxdw [r10-968], r1
+stxdw [r10-848], r3
+mov64 r1, r10
+add64 r1, -136
+mov64 r7, r10
+add64 r7, -1128
+mov64 r2, r7
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
-ldxdw r1, [r10-136]
-stxdw [r10-64], r1
-ldxdw r1, [r10-144]
-stxdw [r10-72], r1
-mov64 r1, r10
-add64 r1, -72
-mov64 r9, 1
+ldxdw r1, [r10-120]
+stxdw [r10-48], r1
+ldxdw r1, [r10-128]
+stxdw [r10-56], r1
+mov64 r1, r10
+add64 r1, -56
+mov64 r8, 1
 mov64 r2, 1
 call anchor_lang::solana_program::log::sol_log_data#9398
-ldxdw r2, [r8+152]
-mov64 r1, r10
-add64 r1, -272
-call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
-mov64 r1, r10
-add64 r1, -224
-mov64 r2, r6
+ldxdw r2, [r6+152]
+mov64 r1, r10
+add64 r1, -256
+call <solana_account_info::AccountInfo as core::clone::Clone>::clone#3106
+mov64 r1, r10
+add64 r1, -208
+mov64 r2, r7
 call <mayhem_program::instructions::pump_interface::sell::SellEvent as anchor_lang::Event>::data#c4f7
 lddw r1, .rodata[e445a52e51cb9a1d]
 lddw r2, .rodata[e445a52e51cb9a1d]
 add64 r2, 8
-stxdw [r10-160], r2
+stxdw [r10-144], r2
+stxdw [r10-152], r1
+ldxdw r1, [r10-208]
 stxdw [r10-168], r1
-ldxdw r1, [r10-224]
+ldxdw r1, [r10-200]
+stxdw [r10-176], r1
 stxdw [r10-184], r1
-ldxdw r1, [r10-216]
-stxdw [r10-192], r1
-stxdw [r10-200], r1
-ldxdw r2, [r10-208]
+ldxdw r2, [r10-192]
 add64 r1, r2
-stxdw [r10-176], r1
-mov64 r1, r10
-add64 r1, -304
+stxdw [r10-160], r1
+mov64 r1, r10
+add64 r1, -288
 mov64 r2, r10
-add64 r2, -200
+add64 r2, -184
 call <alloc::vec::Vec<T> as alloc::vec::spec_from_iter_nested::SpecFromIterNested<T,I>>::from_iter#1a04
-ldxdw r6, [r10-296]
-ldxdw r7, [r10-288]
+ldxdw r6, [r10-280]
+ldxdw r7, [r10-272]
 call _RNvCscR3e5J6VSLN_7___rustc35___rust_no_alloc_shim_is_unstable_v2
 lddw r1, 0x300000000
 ldxdw r2, [r1]
 mov64 r3, 0
 mov64 r1, r2
 add64 r1, -34
-jgt r1, r2, +0x3258
-mov64 r9, 0
-jeq r9, 0, +0x3260
+jgt r1, r2, +0x1c30
+mov64 r8, 0
+jeq r8, 0, +0x1c38
 lddw r1, 0x300007fde
-jeq r2, 0, +0x3280
+jeq r2, 0, +0x1c58
 mov64 r1, r3
 lddw r2, 0x300000008
-jge r1, r2, +0x3298
+jge r1, r2, +0x1c70
 mov64 r1, 1
 mov64 r2, 34
 call alloc::alloc::handle_alloc_error#decd
-jne r9, 0, +0x3208
+jne r8, 0, +0x1be0
 mov64 r3, r1
 lddw r1, 0x300007fde
-jne r2, 0, +0x3220
+jne r2, 0, +0x1bf8
 lddw r2, 0x300000008
-jlt r1, r2, +0x3240
+jlt r1, r2, +0x1c18
 lddw r2, 0x300000000
 stxdw [r2], r1
-ldxdw r2, [r10-272]
+ldxdw r2, [r10-256]
 ldxdw r3, [r2]
 ldxdw r4, [r2+8]
 ldxdw r5, [r2+16]
@@ -1607,22 +888,22 @@
 stxdw [r1+8], r4
 stxdw [r1], r3
 sth [r1+32], 1
-stxdw [r10-64], r1
+stxdw [r10-48], r1
+stdw [r10-40], 1
 stdw [r10-56], 1
-stdw [r10-72], 1
 mov64 r8, r10
-add64 r8, -152
+add64 r8, -136
 mov64 r5, r10
-add64 r5, -72
+add64 r5, -56
 mov64 r1, r8
 lddw r2, .rodata[052ae5d7a7daa724]
 mov64 r3, r6
 mov64 r4, r7
 call solana_instruction::Instruction::new_with_bytes#33e2
 mov64 r6, r10
-add64 r6, -72
+add64 r6, -56
 mov64 r2, r10
-add64 r2, -272
+add64 r2, -256
 mov64 r1, r6
 mov64 r3, 48
 call memcpy
@@ -1636,56 +917,84 @@
 mov64 r1, r0
 lsh64 r1, 32
 rsh64 r1, 32
-jne r1, 26, +0x3410
+jne r1, 26, +0x1ec8
 mov64 r6, 2
-ja +0x3430
+ja +0x1ee8
+mov64 r1, r10
+add64 r1, -1096
+lddw r2, .rodata[07000000]
+call mayhem_program::instructions::pump_interface::buy::BuyError::name#3012
+mov64 r1, r10
+add64 r1, -1072
+mov64 r2, 7
+call <T as alloc::string::SpecToString>::spec_to_string#f398
+lddw r1, .rodata[70726f6772616d73]
+stxdw [r10-1120], r1
+stb [r10-1048], 2
+stw [r10-1104], 97
+stdw [r10-1112], 58
+stdw [r10-1128], 0
+stw [r10-976], 6007
+mov64 r1, r10
+add64 r1, -1128
+call <anchor_lang::error::Error as core::convert::From<anchor_lang::error::AnchorError>>::from#5ca3
+mov64 r2, r1
+ldxdw r1, [r6+16]
+ldxdw r4, [r1+40]
+ldxdw r1, [r6+8]
+ldxdw r3, [r1+136]
+mov64 r1, r0
+call anchor_lang::error::Error::with_values#4132
+ja +0x430
 mov64 r1, r0
 call <anchor_lang::error::Error as core::convert::From<solana_program_error::ProgramError>>::from#507a
 mov64 r6, r0
 mov64 r7, r1
 mov64 r1, r10
-add64 r1, -72
+add64 r1, -56
 call core::ptr::drop_in_place<[solana_account_info::AccountInfo; 1]>#240e
 mov64 r0, r6
 mov64 r1, r7
-jne r0, 2, +0x480
+jne r0, 2, +0x430
 mov64 r0, 2
-ja +0x480
-call abort
-mov64 r9, 2
-ja +0x328
+ja +0x430
+ldxdw r1, [r10-1124]
+stxdw [r10-136], r1
 mov64 r3, r10
-add64 r3, -333
-lddw r1, str("a Display implementation returned an error unexpectedly")
-mov64 r2, 55
-lddw r4, table[3:<core::fmt::Error as core::fmt::Debug>::fmt#aaa5]
-lddw r5, panic_location(/home/runner/work/platform-tools/platform-tools/out/rust/library/alloc/src/string.rs:2830:14)
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:57:23)
+call core::result::unwrap_failed#5a05
+ldxdw r1, [r10-1124]
+stxdw [r10-136], r1
+mov64 r3, r10
+add64 r3, -136
+lddw r1, str("called `Result::unwrap()` on an `Err` value")
+mov64 r2, 43
+lddw r4, table[3:<solana_program_error::ProgramError as core::fmt::Debug>::fmt#4fcd]
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:62:24)
 call core::result::unwrap_failed#5a05
 mov64 r3, r10
-add64 r3, -333
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:350:43)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:79:54)
 call core::result::unwrap_failed#5a05
-stw [r10-1240], 9
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:83:77)
+call core::option::unwrap_failed#2d9b
 mov64 r3, r10
-add64 r3, -1240
-lddw r1, str("called `Result::unwrap()` on an `Err` value")
-mov64 r2, 43
-lddw r4, table[3:<mayhem_program::instructions::pump_amm_interface::pump_swap_types::PumpSwapError as core::fmt::Debug>::fmt#b375]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:378:89)
-call core::result::unwrap_failed#5a05
-mov64 r3, r10
-add64 r3, -333
+add64 r3, -1
 lddw r1, str("called `Result::unwrap()` on an `Err` value")
 mov64 r2, 43
 lddw r4, table[3:<core::num::error::TryFromIntError as core::fmt::Debug>::fmt#ac91]
-lddw r5, panic_location(programs/mayhem-program/src/state.rs:379:34)
+lddw r5, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:83:87)
 call core::result::unwrap_failed#5a05
-lddw r1, panic_location(programs/mayhem-program/src/state.rs:321:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:75:15)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:321:23)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:90:10)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
-lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:322:26)
+lddw r1, panic_location(programs/mayhem-program/src/instructions/pump_interface.rs:90:55)
 call core::panicking::panic_const::panic_const_sub_overflow#7310
